backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java

```diff
@@ -753,57 +756,79 @@ private static Filter convertMapToFilter(String name, Map<String, Object> subFil
+
+      if (subFields.isEmpty()) {
+        // --- Simple leaf field ---
+        result.add(new SelectColumn(column != null ? column.getName() : name));
+
+      } else {
+        // --- Nested or aggregate field ---
+        String suffix =
+            name.endsWith("_agg") ? "_agg" : name.endsWith("_groupBy") ? "_groupBy" : "";
+
+        if (column != null) {
+          TableMetadata refTable = column.isReference() ? column.getRefTable() : null;
+
+          SelectColumn nested =
+              new SelectColumn(
+                  column.getName() + suffix, convertMapSelection(refTable, s.getSelectionSet()));
+
+          // --- Handle filters, limits, offsets, orderby ---
+          if (args != null && !args.isEmpty()) {
             if (args.containsKey(GraphqlConstants.FILTER_ARGUMENT)) {
-              sc.where(
+              nested.where(
                   convertMapToFilterArray(
-                      column.get().getRefTable(),
-                      (Map<String, Object>) args.get(GraphqlConstants.FILTER_ARGUMENT)));
+                      refTable, (Map<String, Object>) args.get(GraphqlConstants.FILTER_ARGUMENT)));
             }
             if (args.containsKey(GraphqlConstants.LIMIT)) {
-              sc.setLimit((int) args.get(GraphqlConstants.LIMIT));
+              nested.setLimit((int) args.get(GraphqlConstants.LIMIT));
             }
             if (args.containsKey(GraphqlConstants.OFFSET)) {
-              sc.setOffset((int) args.get(GraphqlConstants.OFFSET));
+              nested.setOffset((int) args.get(GraphqlConstants.OFFSET));
             }
             if (args.containsKey(GraphqlConstants.ORDERBY)) {
               TableMetadata orderByTable =
-                  column.get().isReference() ? column.get().getRefTable() : column.get().getTable();
-              sc.setOrderBy(convertOrderByIdsToNames(orderByTable, args));
+                  column.isReference() ? column.getRefTable() : column.getTable();
+              nested.setOrderBy(convertOrderByIdsToNames(orderByTable, args));
             }
-            result.add(sc);
-          } else if (agg_fields.contains(s.getName())) {
-            result.add(
-                new SelectColumn(s.getName(), convertMapSelection(aTable, s.getSelectionSet())));
           }
+          result.add(nested);
+
+        } else if (agg_fields.contains(name)) {
+          // --- Aggregate pseudo-field ---
+          result.add(new SelectColumn(name, convertMapSelection(aTable, s.getSelectionSet())));
         }
       }
     }
-    return result.toArray(new SelectColumn[result.size()]);
+
+    return result.toArray(SelectColumn[]::new);
   }
 
+  private static Map<String, Column> buildColumnLookupMap(TableMetadata table) {
+    if (table == null) return Map.of();
+    List<Column> cols = table.getColumnsIncludingSubclasses();
+
+    Map<String, Column> lookup = new HashMap<>(cols.size() * 3);
+    for (Column c : cols) {
+      lookup.put(c.getIdentifier(), c);
```

Comment:
maybe this is overdone. we now make larger map then needed. we could instead simply have helper function that tries getByIdentifier(field), optionally stripping the _agg/_groupBy